File: app/src/main/java/com/OverSadBoy/samurairise/dagger/DBModule.java
package com.OverSadBoy.samurairise.dagger;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;

import com.OverSadBoy.samurairise.model.database.DataBase;

import dagger.Module;
import dagger.Provides;

@Module(includes = ContextModule.class)
public class DBModule {

    @Provides
    DataBase provideDB(Context context) {
        return Room.databaseBuilder(context, DataBase.class, "alarm").build();
    }
}

File: app/src/main/java/com/OverSadBoy/samurairise/dagger/DBModule.kt
package com.OverSadBoy.samurairise.dagger

import android.content.Context
import androidx.test.runner.AndroidJUnit4
import com.OverSadBoy.samurairise.model.database.DataBase
import dagger.Module
import dagger.Provides
import org.junit.runner.RunWith

@Module(includes = [ContextModule::class])
class DBModule {
    @Provides
    fun provideDB(context: Context?): DataBase? {
        return DataBase(context)
    }
}